File: Year 3/React Native/Simple SMHI Weather Warning application/my-second-project/models/Warning.ts
import { Event } from "./Event";
import { AreaName } from "./AreaName";
import { Description } from "./Description";
import { WarningAreas } from "./WarningAreas";

event: Event;  // Placeholder. Please replace with the correct type.
areaName: AreaName;  // Placeholder. Please replace with the correct type.
descriptions: Description;  // Placeholder. Please replace with the correct type.
warningAreas: WarningAreas;  // Placeholder. Please replace with the correct type.

export class Warning {
    id: number;
    event: Event;
    normalProbability: boolean;
    areaName: AreaName;
    descriptions: Description[];
    warningAreas: WarningAreas[];

    constructor() {
        this.id = 0;
        this.event = {} as Event;
        this.normalProbability = false;
        this.areaName = {} as AreaName;
        this.descriptions = [];
        this.warningAreas = [];
    }

    getId() {
        return this.id;
    }
    setId(id:number):void {
        this.id = id;
    }

    getEvent() {
        return this.event;
    }
    setEvent(event: Event): void{
        this.event = event;
    }

    getNormalProbability(){
        return this.normalProbability;
    }
    setNormalProbability(normalProbability: boolean){
        this.normalProbability = normalProbability;
    }

    getAreaName(){
        return this.areaName;
    }
    setAreaName(areaName: AreaName):void{
        this.areaName = areaName;
    }

    getDescriptions(){
        return this.descriptions;
    }
    setDescriptions(descriptions: Description[]):void{
        this.descriptions = descriptions;
    }

    getWarningAreas(){
        return this.warningAreas;
    }
    setWarningAreas(warningAreas: WarningAreas[]):void{
        this.warningAreas = warningAreas;
    }
}

File: Year 3/Android Java/Simple SMHI Warning App JavaVersion/GeoWarning/app/src/main/java/com/example/geowarning/model/Warning.java
package com.example.geowarning.model;

import java.util.List;

public class Warning {
    private int id;
    private Event event;
    private boolean normalProbability;
    private AreaName areaName;
    private List<Description> descriptions;
    private List<WarningAreas> warningAreas;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Event getEvent() {
        return event;
    }

    public void setEvent(Event event) {
        this.event = event;
    }

    public boolean isNormalProbability() {
        return normalProbability;
    }

    public void setNormalProbability(boolean normalProbability) {
        this.normalProbability = normalProbability;
    }

    public AreaName getAreaName() {
        return areaName;
    }

    public void setAreaName(AreaName areaName) {
        this.areaName = areaName;
    }

    public List<Description> getDescriptions() {
        return descriptions;
    }

    public void setDescriptions(List<Description> descriptions) {
        this.descriptions = descriptions;
    }

    public List<WarningAreas> getWarningAreas() {
        return warningAreas;
    }

    public void setWarningAreas(List<WarningAreas> warningAreas) {
        this.warningAreas = warningAreas;
    }
}
